src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2293,12 +2293,145 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
                         {
                             return;
                         }
+
+                    case '>':
+                    case '=':
+                    case '<':
+                        if (!isTrailing)
+                        {
+                            if (IsConflictMarkerTrivia())
+                            {
+                                this.LexConflictMarkerTrivia(ref triviaList);
+                                break;
+                            }
+                        }
+
+                        return;
+
                     default:
                         return;
                 }
             }
         }
 
+        // All conflict markers consist of the same character repeated seven times.  If it is
+        // a <<<<<<< or >>>>>>> marker then it is also followed by a space.
+        private static readonly int s_conflictMarkerLength = "<<<<<<<".Length;
```

Comment:
Minor:  I'd personally make the `"<<<<<<<" a named constant. eg `s_conflictMarker`